spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncherRunner.java

```diff
@@ -45,15 +48,16 @@
 	private ModuleLauncher moduleLauncher;
 
 	@Override
-	public void run(ApplicationArguments applicationArguments) throws Exception {
-		String[] launchedModules = moduleLauncherProperties.getModules();
+	public void run(String... unused) throws Exception {
```

Comment:
My point was exactly to showcase the fact that we're not relying on just the command-line args. I'd settle for something that doesn't even have them if I was sure about the lifecycle (notice how I downgraded from ApplicationRunner already).
I can replace this with a comment if that's ok with you